src/SqlProviderBase.ts

```diff
@@ -576,79 +576,77 @@ class SqlStore implements NoSqlProvider.DbStore {
+                        if (valsRaw) {
+                            const err = _.attempt(() => {
+                                serializedKeys = _.map(NoSqlProviderUtils.arrayify(valsRaw), val =>
+                                    NoSqlProviderUtils.serializeKeyToString(val, <string>index.keyPath));
+                            });
+                            if (err) {
+                                queries.push(SyncTasks.Rejected<void>(err));
+                                return;
                             }
-                        } else {
-                            return;
                         }
+                    } else {
+                        return;
+                    }
 
-                        let valArgs = [], args = [];
-                        _.each(serializedKeys, val => {
-                            valArgs.push('(?, ?)');
-                            args.push(val);
-                            args.push(key);
-                        });
-                        inserts.push(this._trans.internal_nonQuery('DELETE FROM ' + this._schema.name + '_' + index.name +
-                                ' WHERE nsp_refpk = ?', [key]).then(() => {
-                            if (valArgs.length > 0){
-                                return this._trans.internal_nonQuery('INSERT INTO ' + this._schema.name + '_' + index.name +
-                                    ' (nsp_key, nsp_refpk) VALUES ' + valArgs.join(','), args);
-                            }
-                        }));
+                    let valArgs = [], args = [];
+                    _.each(serializedKeys, val => {
+                        valArgs.push('(?, ?)');
+                        args.push(val);
+                        args.push(key);
                     });
-                    return SyncTasks.all(inserts).then(_.noop);
+                    queries.push(this._trans.internal_nonQuery('DELETE FROM ' + this._schema.name + '_' + index.name +
+                            ' WHERE nsp_refpk = ?', [key]));
+                    if (valArgs.length > 0){
+                        queries.push(this._trans.internal_nonQuery('INSERT INTO ' + this._schema.name + '_' + index.name +
+                            ' (nsp_key, nsp_refpk) VALUES ' + valArgs.join(','), args));
+                    }
                 });
-
-                return SyncTasks.all(queries).then(_.noop);
+            });
+            if (err) {
+                return SyncTasks.Rejected<void>(err);
             }
-        });
+        }
+
+        let promise = SyncTasks.all(queries);
         if (this._verbose) {
             promise = promise.finally(() => {
                 console.log('SqlStore (' + this._schema.name + ') put: (' + (Date.now() - startTime) + 'ms): Count: ' + items.length);
             });
         }
-        return promise;
+        return promise.then(_.noop);
```

Comment:
SyncTasks.all typings will return a promise with array return value.  _.noop changes return type back to undefined